improve add-ons pages for better integration discovery (#5499)
withastro/docs: src/content/docs/en/guides/server-side-rendering.mdx

For example, to install the Vercel adapter manually:

1. Install the adapter to your project dependencies using your preferred package manager:

   <PackageManagerTabs>
     <Fragment slot="npm">
     ```shell
     npm install @astrojs/vercel
     ```
     </Fragment>
     <Fragment slot="pnpm">
     ```shell
     pnpm install @astrojs/vercel
     ```
     </Fragment>
     <Fragment slot="yarn">
     ```shell
     yarn add @astrojs/vercel
     ```
     </Fragment>
   </PackageManagerTabs>

2. [Add the adapter](/en/reference/configuration-reference/#adapter) to your `astro.config.mjs` file's import and default export, along with your desired `output` mode:

    ```js ins={3,7} {6}
    // astro.config.mjs
    import { defineConfig } from 'astro/config';
    import vercel from '@astrojs/vercel/serverless';

    export default defineConfig({
      output: 'server',
      adapter: vercel(),
    });
    ```

    Note that different adapters may also have different configuration settings. Read each adapter's documentation, and apply any necessary config options to your chosen adapter in `astro.config.mjs`

### Configure `server` or `hybrid`

> To enable on-demand rendering, you will need to update your `output` configuration to one of the two server-rendered modes.

Reviewer: ```suggestion
To enable on-demand rendering, you must update your `output` configuration to one of the two server-rendered modes.
```

Just a small nit here, instead of "do it later," "do it now!"

Probably meaningless, just my two cents! :)

Author: Sure, I'll take it!
